src/main/java/com/snowflake/kafka/connector/internal/streaming/TopicPartitionChannel.java

```diff
@@ -170,6 +176,36 @@ public class TopicPartitionChannel {
   // Used to identify when to flush (Time, bytes or number of records)
   private final BufferThreshold streamingBufferThreshold;
 
+  // Whether schematization has been enabled.
+  private boolean enableSchematization;
+
+  // Whether schema evolution could be done on this channel
+  private boolean enableSchemaEvolution = false;
+
+  private SnowflakeConnectionService conn;
+
+  /** Initialize TopicPartitionChannel without the connection service */
+  public TopicPartitionChannel(
+      SnowflakeStreamingIngestClient streamingIngestClient,
+      TopicPartition topicPartition,
+      final String channelName,
+      final String tableName,
+      final BufferThreshold streamingBufferThreshold,
+      final Map<String, String> sfConnectorConfig,
+      KafkaRecordErrorReporter kafkaRecordErrorReporter,
+      SinkTaskContext sinkTaskContext) {
+    this(
+        streamingIngestClient,
+        topicPartition,
+        channelName,
+        tableName,
+        streamingBufferThreshold,
+        sfConnectorConfig,
+        kafkaRecordErrorReporter,
+        sinkTaskContext,
+        null);
```

Comment:
This is not required for this PR but:
I do feel this can be refactored and we dont need to pass in connectionService. 

Consider creating a class called SchemaManager which can be created from SnowflakeSinkServiceV2 and it will have enableSchemaMapping, enableSchemaEvolution and connectionService instance. 

All the helper methods related to schematization can reside in that and we can avoid creating a huge monolith like TopicPartitionChannel. 

Please file a JIRA for this.